falcon/request.py

```diff
@@ -619,11 +613,30 @@ def date(self):
 
     @property
     def if_match(self):
-        return helpers.parse_etags(self.env.get('HTTP_IF_MATCH'))
+        # TODO(kgriffs): It may make sense at some point to create a
+        #   header property generator that DRY's up the memoization
+        #   pattern for us.
+        # PERF(kgriffs): It probably isn't worth it to set
+        #   self._cached_if_match to a special type/object to distinguish
+        #   between the variable being unset and the header not being
+        #   present in the request. The reason is that if the app
+        #   gets a None back on the first reference to property, it
+        #   probably isn't going to access the property again (TBD).
+        if self._cached_if_match is None:
+            header_value = self.env.get('HTTP_IF_MATCH')
+            if header_value:
```

Comment:
This `if` check seems to be a bit superfluous, since `_parse_etags` can understand `None`, and explicitly so by design:
```python
if etag_str is None:
        return None
```

Or is it by design in order not to add an attribute needlessly?
However, in that case the snippet above is not really needed in `_parse_etags`.

Reply:
This is a bit of cruft left over from the refactor. The if statement is an optimization to avoid both a function call and setting an attribute. All things considered, I think it probably makes sense to just remove the statement inside `_parse_etags()` that you quoted above.

Reply:
Updated. Let me know what you think.

Reply:
Looking good now IMHO!